### PR TITLE
Add suggest improvement button to vulnerability page

### PR DESCRIPTION
**Pull request by ZhangChen199102**
Issue: #1005

Result:
image

image

### CODE REVIEW

**Comment by andrewpollock** [comment]
I think suggesting improvements from a mobile device is probably going to be a dreadful user experience, so I suggest not presenting the button on small screens.

For large screens, could we try utilizing the left margin for actions? Bring the button up to be beside the **Source** field.
For

File: gcp/appengine/frontend3/src/templates/vulnerability.html [hunk 1]
```diff
@@ -221,6 +221,14 @@ <h3 class="mdc-layout-grid__cell--span-3">
   </div>
   {% endfor -%}
   </spicy-sections>
+
+  {% if vulnerability.human_source_link and vulnerability.human_source_link.startswith("https://github.com/advisories/") %}
```
Comment (andrewpollock) [inline gcp/appengine/frontend3/src/templates/vulnerability.html:225]: Please add an else case for the other sources that creates a button titled "See a problem?" that links to https://google.github.io/osv.dev/faq/#ive-found-something-wrong-with-the-data

File: gcp/appengine/frontend3/src/templates/vulnerability.html [hunk 2]
```diff
@@ -283,6 +283,14 @@ <h3 class="mdc-layout-grid__cell--span-3">
       </div>
       {% endfor -%}
     </spicy-sections>
+  
+    {% if vulnerability.human_source_link and vulnerability.human_source_link.startswith("https://github.com/advisories/") -%}
```
Comment (ZhangChen199102) [inline gcp/appengine/frontend3/src/templates/vulnerability.html:287]: I found that not every vulnerability has a github advisory (e.g.https://osv.dev/vulnerability/GO-2024-2571), so I think we might want to check if the `human_source_link` starts with 'https://github.com/advisories/'. Please let me know if this is wrong.

**Comment by ZhangChen199102** [comment]
> I think suggesting improvements from a mobile device is probably going to be a dreadful user experience, so I suggest not presenting the button on small screens.
>
>For large screens, could we try utilizing the left margin for actions? Bring the button up to be beside the Source field.
For

Thanks @andrewpollock, I've tried moving the button next to Source, but I found the UI is bit of Inconsistent (probably because the height of the button):
image

So I tried moving below of Source and it looks better I think:
image

Please let me know your thought

**Comment by andrewpollock** [comment]
I was actually thinking about having it in the left gutter next to the Source title text

**Comment by andrewpollock** [comment]
I was actually thinking about having it in the left gutter next to the Source title text itself, not the link

**Comment by ZhangChen199102** [comment]
@andrewpollock do you mean something like this:
WX20240527-100619@2x

Or maybe:
left-side-button

**Comment by andrewpollock** [comment]
@ZhangChen199102 yes, the second option in https://github.com/google/osv.dev/pull/2227#issuecomment-2132519497 was what I had in mind. 

You'll need to adjust the overall height of each row so that when the button gets rendered it doesn't make the Source line vertical spacing look unusual.

Another possibility is to de-couple it from the Source row and put the button to the left of the identifier instead...

**Comment by ZhangChen199102** [comment]
@andrewpollock thanks, my only concern is when the screen gets narrower (but still wider than mobile devices screen), there will be very limited space between the text body and browser's border, so I wonder where should we put the button in this case:
image

**Comment by andrewpollock** [comment]
> my only concern is when the screen gets narrower

Is it possible to progressively not render components in this case? Or if we go with putting the button to the side of the identifier, it could go on the right hand side of the identifier at all times. There's guaranteed to be plenty of screen real estate on that side...

**Comment by ZhangChen199102** [comment]
I've moved the button next to the title:
image

Also I can move the button to very right side if you think they're too closed, e.g.
image

**Comment by andrewpollock** [comment]
I like the right-justified button (next to the ID) best of what I've seen so far.

**Comment by andrewpollock** [comment]
We talked about this in our team meeting, and agree that the right-justified button on the same line as the identifier is preferred. We would like to de-emphasize the button a little, and were thinking about the ecosystem buttons on https://osv.dev/list

**Comment by ZhangChen199102** [comment]
Thanks @andrewpollock, here's the latest result:
image

And the button will be hidden for mobile device:
image

**Review by andrewpollock — APPROVED** [review]
LGTM!

**Comment by ZhangChen199102** [comment]
@another-rex here’s the button without border and background color:
image

image

**Comment by another-rex** [comment]
Let's go with the link option (no border/bg color)!

**Pull request merged**